pilot/pkg/xds/endpoint_builder.go

```diff
@@ -414,29 +414,25 @@ func buildEnvoyLbEndpoint(b *EndpointBuilder, e *model.IstioEndpoint) *endpoint.
 	address, port := e.Address, e.EndpointPort
 	tunnelAddress, tunnelPort := address, model.HBoneInboundListenPort
 
-	supportsTunnel := false
+	requestsTunnel := false
 	// Other side is a waypoint proxy.
 	if al := e.Labels[constants.ManagedGatewayLabel]; al == constants.ManagedGatewayMeshControllerLabel {
-		supportsTunnel = true
+		requestsTunnel = true
 	}
 
 	// Otherwise has ambient enabled. Note: this is a synthetic label, not existing in the real Pod.
-	if b.push.SupportsTunnel(e.Address) {
-		supportsTunnel = true
+	if b.push.RequestsTunnel(e.Address) {
+		requestsTunnel = true
 	}
 	// Otherwise supports tunnel
 	// Currently we only support HTTP tunnel, so just check for that. If we support more, we will
 	// need to pick the right one based on our support overlap.
-	if e.SupportsTunnel(model.TunnelHTTP) {
-		supportsTunnel = true
+	if e.RequestsTunnel(model.TunnelHTTP) {
+		requestsTunnel = true
 	}
 	if b.proxy.IsProxylessGrpc() {
 		// Proxyless client cannot handle tunneling, even if the server can
-		supportsTunnel = false
-	}
-
-	if !b.proxy.EnableHBONE() {
```

Comment:
I think this is still needed or we will use HBONE when its not enabled?

Reply:
I made it so that the label also implies listen.

Reply:
Not sure I follow. It seems like features.PILOT_ENABLE_HBONE=false can still have requestsTunnel=true? which is what I am trying to avoid. It doesn't matter if the label is there, if we don't enable HBONE it must not use hbone

Reply:
Enable is too broad a term so I'm not sure what you mean by that. During gradual transition, workloads will listen but not send to tunnel ports. Then they can start sending and validate conformance.

Reply:
`features.PILOT_ENABLE_HBONE=false` means "do not give me ANYTHING hbone related".

Reply:
Yeah I got it - added a clause.